sys/compat/cheriabi/cheriabi.h

```diff
@@ -66,18 +66,18 @@ cheriabi_cap_to_ptr(caddr_t *ptrp, void * __capability cap, size_t reqlen,
 	if (!cheri_gettag(cap)) {
 		if (!may_be_null)
 			return (EFAULT);
-		*ptrp = (caddr_t)cap;
+		*ptrp = (__cheri_cast void *)cap;
```

Comment:
I believe this wants to be
```
*ptrp = (void *)(vaddr_t)cap;
```
or maybe even
```
*ptrp = (caddr_t)(uintptr_t)(cheri_getbase(cap) + cheri_getoffset(cap));
```
per CTSRD-CHERI/clang#152

Reply:
Good catch, I think I'll go for the latter as it's even more explicit. Or maybe vaddr_t plus a comment linking to the issue.